Assets/MixedRealityToolkit.SDK/Experimental/Features/Input/Scripts/InputSystemManager.cs

```diff
@@ -0,0 +1,104 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using Microsoft.MixedReality.Toolkit.Utilities;
+using UnityEngine;
+using UnityEngine.EventSystems;
+
+#if UNITY_EDITOR
+using Microsoft.MixedReality.Toolkit.Input.Editor;
+#endif
+
+namespace Microsoft.MixedReality.Toolkit.Input
+{
+    /// <summary>
+    /// Service manager supporting running the input system, without requring the MixedRealityToolkit object.
+    /// </summary>
+    public class InputSystemManager : BaseServiceManager
+    {
+        [SerializeField]
+        [Tooltip("The input system type that will be instantiated.")]
+        [Implements(typeof(IMixedRealityInputSystem), TypeGrouping.ByNamespaceFlat)]
+        private SystemType InputSystemType = null;
+
+        [SerializeField]
+        [Tooltip("The input system configuration profile.")]
+        private MixedRealityInputSystemProfile profile = null;
+
+        private void Awake()
+        {
+            InitializeManager();
+        }
+
+        protected override void OnDestroy()
+        {
+            UninitializeManager();
+            base.OnDestroy();
+        }
+
+        /// <inheritdoc />
+        public override bool RegisterDataProvider<T>(T dataProviderInstance)
+        {
+            bool registered = base.RegisterDataProvider<T>(dataProviderInstance);
+            if (registered)
+            {
+                dataProviderInstance.Initialize();
```

Comment:
>dataProviderInstance.Initialize(); [](start = 16, length = 34)

Do you have any guidance on why the input system has to call this as part of register data provider, but this isn't part of the base register data provider implementation?

Reply:
you know, that should likely be pushed into the base class... good catch!

Reply:
fixed